cdk/.gitignore

```diff
@@ -1,2 +1,6 @@
 cdk.out/
 *.js
+
+# Ensure that this package-lock.json file is checked in properly regardless
+# of any settings at a more global level in the repository.
+!package-lock.json
```

Comment:
Neat trick.

Reply:
Yeah it's super helpful when:
 - you need to write an overly-broad rule (for example to ignore all `.js` files) but you need a particular one or two (for config or somethng)
 - you have a rule in a parent directory that you need to not apply in a child directory